internal/repositories/repositories.go

```diff
@@ -99,6 +115,26 @@ func (r *Repository) GetSignalLatest(ctx context.Context, latestArgs *model.Late
 	return coll, nil
 }
 
+// GetAvailableSignals returns the available signals for the given tokenID and filter.
+// If no signals are found, a nil slice is returned.
+func (r *Repository) GetAvailableSignals(ctx context.Context, tokenID uint32, filter *model.SignalFilter) ([]string, error) {
+	allSignals, err := r.chService.GetAvailableSignals(ctx, tokenID, filter)
+	if err != nil {
+		return nil, handleDBError(err, r.log)
+	}
+	for i, signal := range allSignals {
+		// remove signals that are not queryable
+		if _, ok := r.queryableSignals[signal]; !ok {
+			allSignals = slices.Delete(allSignals, i, i+1)
+		}
+	}
```

Comment:
[`DeleteFunc`](https://pkg.go.dev/slices#DeleteFunc)? You wouldn't have to futz with indices then.

Reply:
I don't love `DeleteFunc` because declaring a new function seems overcomplicated. Plus, it also forces you to iterate over the entire list and call the function for each element. I have similar thoughts about even using `slices.Delete` over `append(s[:i], s[i+1]...)`

Reply:
Actually, does this work? Don’t the indices shift once you delete?

Reply:
I would agree with you in most cases but `i, i+1` has always been the recommended way of removing elements from a list in golang. https://go.dev/wiki/SliceTricks

Reply:
https://go.dev/play/p/3pvvgpTtx90

Reply:
Yeah, I think that is a separate issue from deleting it in a range loop without adjusting the iterator. I am just going to create a new list to keep things simple.